docs/developer/architecture/ci.rst

```diff
@@ -0,0 +1,122 @@
+CI
+==
+
+Pre-merge
+---------
+
+What should be tested in pre-merge on every ``user/*``, ``feature/*``,
+``improvement/*``, ``bugfix/*``, ``w/*`` branches. The pre-merge test should
+not long too much time (less than 40 minutes is great) so we can't
+test everything in pre-merge, we should only test building of the product
+and check that product still usable.
+
+- Building tests
+
+  - Build
+  - Lint
+  - Unit tests
+
+- Installation tests
+
+  - Simple install RHEL
+  - Simple install CentOs + expansion
+
+When merging several pull requests at the same time, given that we are on a
+queue branch (``q/*``), we may want to have or we may require additional tests
+as combinaison of several PRs could have larger impact than all individual PR.
+
+- Simple upgrade/downgrade
+
+.. _post-merge:
+
+Post-merge
+----------
+
+On each and every ``development/2.*`` branches we want to run complex tests
+to ensure that the current product continues to work.
+
+Nightly
+^^^^^^^
+
+- Upgrade, downgrade tests:
+
+  - For development branch just before
+
+    e.g.: on ``development/2.x`` test upgrade from ``development/2.(x-1)``
+    and downgrade to ``development/2.(x-1)``
+
+    - Build branch ``development/2.(x-1)``
```

Comment:
It's not released version so we can't rely on artifacts as we don't know if we already have one or not

Reply:
It's post-merge, so we have the pre-merge artifacts available ;)

Reply:
No not necessarily post-merge should run even if nothing is merged on the branch since long time to check that product is not broken due to dependency ... (that's why also we should re-built it) 
Anyway artifacts get removed after ... some time so for previous version we don't know if we will have one available or not

Reply:
Mmmh, still not really convinced by the need to rebuild in post-merge. The product dependencies are checked at runtime, when we install. Having a broken build will be visible whenever we run any pre-merge, and if you're really scared that it breaks on some old, inactive branch, you can run a build stage for the current branch in post merge. But doing it twice for each branch (when it's `N` and `N-1`), really not convinced by the benefit (vs. the fact that we'd loose the rest of the tests if this one fails).
W.r.t maintaining build artifacts for old dev branches, I'd argue this is not hard to remedy (define a persistent location for "latest dev/" builds, re-build only when there are no artifacts available, ...).

Reply:
> W.r.t maintaining build artifacts for old dev branches, I'd argue this is not hard to remedy (define a persistent location for "latest dev/" builds, re-build only when there are no artifacts available, ...).

Yes it's doable, but except if we have something directly in `bert-e` to keep the last build of the dev branches, we may need to build

Reply:
Keeping this conversation open to track when we start writing tickets about this ;)

Reply:
We already have this one #2222